tsl/src/continuous_aggs/create.c

```diff
@@ -1460,10 +1539,21 @@ add_aggregate_partialize_mutator(Node *node, AggPartCxt *cxt)
 		if (cxt->ignore_aggoid == ((Aggref *) node)->aggfnoid)
 			return node; /*don't process this further */
 
+		bool old_inside_aggref = cxt->inside_aggref;
+		if (!old_inside_aggref)
+			cxt->inside_aggref = true;
+
 		Aggref *newagg = add_partialize_column((Aggref *) node, cxt);
```

Comment:
Not a fan of declaring variables in the middle lines like this. Even though this is existing code, we could clean this up. Also, shouldn't the compiler have barfed about this?

Reply:
Our codebase is full with this kind of declarations. This is because we follow C99 standards. See our [style guide](https://github.com/timescale/timescaledb/blob/master/docs/StyleGuide.md#declarations-before-code).